app/code/Magento/Catalog/Model/Indexer/Category/Flat/AbstractAction.php

```diff
@@ -77,20 +76,31 @@ class AbstractAction
      */
     protected $skipStaticColumns = [];
 
+    /**
+     * @var SkipStaticColumnsProvider
+     */
+    private $staticColumnsProvider;
```

Comment:
Perhaps it should be named `$skipStaticColumnsProvider`
```suggestion
    private $skipStaticColumnsProvider;
```